cmd/weaver/types.go

```diff
@@ -23,6 +25,27 @@ type argument struct {
 	ArrayLength    int // Set as 0 if not array
 }
 
+type procInfo struct {
+	Pid  uint32 `json:"pid,omitempty"`
+	Ppid uint32 `json:"ppid,omitempty"`
+	Comm string `json:"comm,omitempty"`
+}
+
+// UnmarshalBinary for procInfo
+func (i *procInfo) UnmarshalBinary(data []byte) error {
+
+	// proc info struct is 24 bytes long
+	if len(data) < 24 {
```

Comment:
Can check `if len(data) == 24` instead

Reply:
This was my first try, but somehow the byte slice isn't 24 bytes long... Even though the C struct should be 24 bytes long (but i'm no expert). There is more bytes sent than what we expect

Reply:
Ah yea, it's the padding '\x00's that you're removing below. Like all other data, you're sending over literal raw bytes. 

I bet if you put this check after the `bytes.Trim` call you can check `if len(data) == 24`. Not at a computer that I can run this at the moment, so just a theory.

Reply:
Indeed it worked by placing the `bytes.Trim` before the check

Reply:
In fact checking if len != 24 does not work, `bytes.Trim` also delete the padding '\x00' in the 16 bytes comm buffer. Instead, I did this 
```
	// proc info struct is 24 bytes long and should at least be 8 bytes long
	if len(data) > 24 || len(data) < 8 {
		return fmt.Errorf("error decoding process info")
	}
```